### PR TITLE
refactor(maven): Refactor datasource tests

### PR DESCRIPTION
**Pull request by zharinov**
## Changes:

Use and track HTTP/file mocks instead of relying on file protocol support being used to obtain fixtures.

## Context:

Ref: #6742

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [x] Newly added unit tests, or
- [ ] No new tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

File: lib/datasource/maven/util.ts [hunk 1]
```diff
@@ -189,6 +189,7 @@ export async function getDependencyInfo(
 
   const pomUrl = getMavenUrl(dependency, repoUrl, path);
   const { xml: pomContent } = await downloadMavenXml(pomUrl);
+  // istanbul ignore if
```
Comment (zharinov) [inline lib/datasource/maven/util.ts:192]: Will fix it in separate PR, need to wrap XML parsing in try-catch (which is behavioral change).

File: lib/datasource/maven/index.spec.ts [hunk 2]
```diff
@@ -1,314 +1,315 @@
-import nock from 'nock';
-import { Release, getPkgReleases } from '..';
-import { getName, loadFixture } from '../../../test/util';
+import _fs from 'fs-extra';
```
Comment (viceice) [inline lib/datasource/maven/index.spec.ts:1]: Isn't the code using our `utils/fs` wrapper?

Comment (zharinov) [reply lib/datasource/maven/index.spec.ts:1]: Won't work since `maven/util.ts` uses `fs-extra` directly

Comment (viceice) [reply lib/datasource/maven/index.spec.ts:1]: ok, so needs refactored /fixed in separate pr.

File: lib/datasource/maven/__snapshots__/index.spec.ts.snap [hunk 3]
```diff
@@ -339,7 +339,7 @@ Array [
       "user-agent": "https://github.com/renovatebot/renovate",
     },
     "method": "GET",
-    "url": "https://repo.maven.apache.org/maven2/org/example/package/maven-metadata.xml",
+    "url": "http://repo.maven.apache.org/maven2/org/example/package/maven-metadata.xml",
```
Comment (rarkins) [inline lib/datasource/maven/__snapshots__/index.spec.ts.snap:342]: ??

Comment (zharinov) [reply lib/datasource/maven/__snapshots__/index.spec.ts.snap:342]: Intentionally replaced `https` to `http` for this test. I needed to fix coverage and this test seemed pretty relevant for this.

Comment (zharinov) [reply lib/datasource/maven/__snapshots__/index.spec.ts.snap:342]: Instead, probably it would be better to remove HTTP support at all

Comment (twendelmuth) [reply lib/datasource/maven/__snapshots__/index.spec.ts.snap:342]: Wouldn't removing HTTP support break support for internal maven repositories that aren't served via HTTPS? 
Like sonatype nexus for example still comes out of the box with http as it's default port.
Other scenarios might see some old references to http:// where the server properly handles a 301 to https

Comment (zharinov) [reply lib/datasource/maven/__snapshots__/index.spec.ts.snap:342]: Yes, you're right

**Review by rarkins — APPROVED** [review]

**Pull request merged**